File: features/document/src/main/java/com/nivasafinance/features/document/config/DocumentStorageProperties.java
package com.nivasafinance.features.document.config;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "document.storage")
@Data
public class DocumentStorageProperties {
    
    private String provider = "LOCAL";
    private SignedUrlProperties signedUrl = new SignedUrlProperties();
    private FallbackProperties fallback = new FallbackProperties();
    
    @Data
    public static class SignedUrlProperties {
        private Long expirationSeconds = 300L;
    }
    
    @Data
    public static class FallbackProperties {
        private Boolean enabled = true;
    }
}

File: features/document/src/main/java/com/nivasafinance/features/document/config/DocumentStorageProperties.kt
package com.nivasafinance.features.document.config

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "document.storage")
data class DocumentStorageProperties(
    val provider: String = "LOCAL",

    val signedUrl: SignedUrlProperties = SignedUrlProperties(),

    val fallback: FallbackProperties = FallbackProperties()
)

data class SignedUrlProperties(
    val expirationSeconds: Long = 300L
)

data class FallbackProperties(
    val enabled: Boolean = true
)

@ConfigurationProperties(prefix = "aws.s3")
data class S3Properties(
    val region: String = "",

    val bucketName: String = "",

    val useIamRoles: Boolean = false,

    val accessKeyId: String = "",

    val secretAccessKey: String = ""
)

@ConfigurationProperties(prefix = "local.storage")
data class LocalStorageProperties(
    val basePath: String = "/tmp/documents"
)
